### DISCUSSION TITLE
Outsource face recognition into docker container

### DISCUSSION THREAD

**Discussion by qudiqudi** [nextcloud/all-in-one#2025] [Ideas] [answered]
As it came clear to me and many others, at the moment, the capabilities of nextcloud with facial and/or object recognition are limited to a few dependencies often unavailable due to system or architecture limitation.

Would it be possible to outsource the recognition functions into a seperated docker container with backed in dependencies that can communicate with nextcloud?
Just like imagninary "outsourced" the preview generation.

There are already a few approaches of facial recognition inside docker containers available that can communicate via websockets.
https://github.com/SthPhoenix/InsightFace-REST
https://github.com/JanLoebel/face_recognition
https://github.com/ageitgey/face_recognition

With this, users had the option to opt-in this feature via the Master-Container-GUI like Collabora or AV.
My technical knowledge is not enough to understand whether this is even possible.

**Comment by szaimen** [comment, marked as answer]
Hi,

as for recognize, I would gladly add an additional container after https://github.com/nextcloud/recognize/issues/73 is implemented.

As for facerecognition, you should be looking forward https://github.com/nextcloud/all-in-one/discussions/1581 which will allow to add additional container that the community can make and maintain.
